app/forms/FormsProvider.scala

```diff
@@ -46,28 +42,13 @@ class FormsProvider() {
     missingInputError = "common.endDateQuestionPage.error", Seq(taxYear.toString)
   )
 
-  def endDateForm(taxYear: Int,
-                  benefitType: BenefitType,
-                  isAgent: Boolean,
-                  claimStartDate: LocalDate)
-                 (implicit messages: Messages): Form[DateFormData] = {
-    lazy val isAgentSuffix = userType(isAgent)
-    val emptyDayKey = s"${benefitType.typeName}.endDatePage.error.empty.day.$isAgentSuffix"
-    val emptyMonthKey = s"${benefitType.typeName}.endDatePage.error.empty.month.$isAgentSuffix"
-    val emptyYearKey = s"${benefitType.typeName}.endDatePage.error.empty.year.$isAgentSuffix"
-    val invalidDateKey = s"${benefitType.typeName}.endDatePage.error.invalid.date"
-
-    Form(
-      dateMapping(emptyDayKey, emptyMonthKey, emptyYearKey, invalidDateKey, None)
-        .verifying(
-          messages(s"${benefitType.typeName}.endDatePage.error.mustBeEndOfYear.$isAgentSuffix", (taxYear - 1).toString, taxYear.toString),
-          dateFormData => dateFormData.toLocalDate
-            .forall(date => date.isAfter(LocalDate.of(taxYear - 1, APRIL, SIX - 1)) && date.isBefore(LocalDate.of(taxYear, APRIL, SIX)))
-        ).verifying(
-        messages(s"${benefitType.typeName}.endDatePage.error.mustBeAfterStartDate.$isAgentSuffix", translatedDateFormatter(claimStartDate)),
-        dateFormData => dateFormData.toLocalDate.forall(date => date.isAfter(claimStartDate))
-      )
-    )
```

Comment:
After removing that you should remove dateMapping from MappingUtil since it is not going to be used anymore.